src/main/java/com/github/grignaak/collections/CowCollectors.java

```diff
@@ -0,0 +1,49 @@
+package com.github.grignaak.collections;
+
+import java.util.Comparator;
+import java.util.Set;
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collector.Characteristics;
+
+@SuppressWarnings("WeakerAccess")
+public final class CowCollectors {
+    public static <T> Collector<T, ?, CowList<T>> toCowList() {
+        return Collectors.toCollection(CowArrayList::new);
+    }
+
+    public static <T> Collector<T, ?, CowSet<T>> toCowSet() {
+        BinaryOperator<CowSet<T>> combiner = (left, right) -> {
```

Comment:
This can simply be Collectors.toCollection(CowHashSet::new)

Reply:
are you sure? i think "unordered" characteristic can help improving perfomance of "collect" operations in case of some streams

Reply:
That's a good point, and it doesn't make this code that much less unreadable.